### PR TITLE
Ur mom

### LINKED ISSUE #35: YourMom
Obligatory YourMom subsystem and command.

### PR DESCRIPTION
**Pull request by Jack-Haefele**
closes #35

### CODE REVIEW

**Review by QuackitsQuinn — CHANGES_REQUESTED** [review]
I think you need to add more UrMom into the UrMom command

File: src/main/java/frc/robot/commands/DeployUrMom.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  public void execute() {}
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {}
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
+    return false;
```
Comment (willitcode) [inline src/main/java/frc/robot/commands/DeployUrMom.java:44]: `return true` so that the command will end sometime.

**Review by willitcode — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/DeployUrMom.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  /** Creates a new DeployUrMom. */
+  public DeployUrMom() {
+    this.urmom = new UrMom();
+    addRequirements(urmom);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    System.out.println("your mom");
```
Comment (Berdenson) [inline src/main/java/frc/robot/commands/DeployUrMom.java:30]: This should be a method in UrMom

File: src/main/java/frc/robot/commands/DeployUrMom.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.UrMom;
+
+/** Prints "your mom" */
+public class DeployUrMom extends Command {
+  private UrMom urmom;
+
+  /** Creates a new DeployUrMom. */
+  public DeployUrMom() {
+    this.urmom = new UrMom();
```
Comment (willitcode) [inline src/main/java/frc/robot/commands/DeployUrMom.java:23]: Don't make a new instance of a subsystem inside a command—all subsystems should be only ever be initialized once: in RobotContainer

File: src/main/java/frc/robot/commands/DeployUrMom.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    addRequirements(urmom);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {}
```
Comment (willitcode) [inline src/main/java/frc/robot/commands/DeployUrMom.java:29]: You'll want to call some method in the subsystem (an example might be `urmom.printUrMom()` here

File: src/main/java/frc/robot/subsystems/UrMom.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class UrMom extends SubsystemBase {
+  /** Creates a new UrMom. */
+  public UrMom() {
+    System.out.println("your mom");
```
Comment (willitcode) [inline src/main/java/frc/robot/subsystems/UrMom.java:19]: Don't do this here. Instead, make another method—an example header might be `public void printUrMom()`—that prints "your mom"

**Review by willitcode — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/UrMom.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+ * The First Law: A robot may not injure a human being or, through inaction, allow a human being to come to harm.
+ * The Second Law: A robot must obey the orders given it by human beings except where such orders would conflict with the First Law.
+ * The Third Law: A robot must protect its own existence as long as such protection does not conflict with the First or Second Law.
+ */
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class UrMom extends SubsystemBase {
```
Comment (Berdenson) [inline src/main/java/frc/robot/subsystems/UrMom.java:16]: you need javadoc explaining the purpose of this subsystem

**Review by Berdenson — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/DeployUrMom.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.RobotContainer;
+import frc.robot.subsystems.UrMom;
+
+/** Prints "your mom" */
+public class DeployUrMom extends Command {
+  private UrMom urmom;
```
Comment (Berdenson) [inline src/main/java/frc/robot/commands/DeployUrMom.java:20]: Improperly capitalized object. Please rename to urMom to fit with the rest of the code's formatting.

**Review by Berdenson — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/RobotContainer.java [hunk 8]
```diff
@@ -90,6 +96,7 @@ private void configureBindings() {
     driverController.x().onTrue(enterXMode);
     driverController.a().onTrue(runIntake);
     driverController.y().onTrue(deployPneumatics);
+    driverController.rightTrigger().onTrue(deployUrMom);
```
Comment (willitcode) [inline src/main/java/frc/robot/RobotContainer.java:99]: ...tbh let's not bind a button to run this command. We'll just wind up unbinding it later when we inevitably need that button for something actually useful

**Review by willitcode — CHANGES_REQUESTED** [review]

**Review by willitcode — APPROVED** [review]

**Comment by Jack-Haefele** [comment]
you know who else

**Review by Berdenson — APPROVED** [review]

**Review by QuackitsQuinn — APPROVED** [review]
still needs more ur moms but i guess its fine....

**Pull request merged**